subsys/bluetooth/mesh/main.c

```diff
@@ -163,18 +161,20 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 		bt_mesh_net_store();
 	}
 
-	bt_mesh_start();
+	atomic_set_bit(bt_mesh.flags, BT_MESH_VALID);
+
+	return bt_mesh_start();
```

Comment:
I guess since bt_mesh_start can fail, we should not return its error, otherwise it will be treated by provisionee as a failure:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys%2Fbluetooth%2Fmesh%2Fprovisionee.c#L595-L606